integration_tests/src/main/python/spark_session.py

```diff
@@ -145,3 +145,7 @@ def is_spark_330_or_later():
 def is_databricks91_or_later():
     spark = get_spark_i_know_what_i_am_doing()
     return spark.conf.get("spark.databricks.clusterUsageTags.sparkVersion", "") >= "9.1"
+
+def is_databricks104_or_later():
+    spark = get_spark_i_know_what_i_am_doing()
+    return spark.conf.get("spark.databricks.clusterUsageTags.sparkVersion", "") >= "10.4"
```

Comment:
This string comparison doesn't work as a numeric comparison.

Reply:
You could zero lpad the spark conf.

Reply:
Interesting, but it doesn't address the same issue happening in the minor version number.  I added code to split on `.` and convert the first two fields to integers which should resolve it.